packages/ui/react-ui-editor/src/extensions/dnd.ts

```diff
@@ -3,15 +3,23 @@
 //
 
 import type { Extension } from '@codemirror/state';
-import { EditorView } from '@codemirror/view';
+import { dropCursor, EditorView } from '@codemirror/view';
 
-import { log } from '@dxos/log';
+export type DNDOptions = { onDrop?: (view: EditorView, event: { files: FileList }) => void };
 
-// TODO(burdon): DND experiment.
-export const dnd = (): Extension => {
-  return EditorView.domEventHandlers({
-    drop: (event, view) => {
-      log.info('domEventHandlersDrop', { event });
-    },
-  });
+export const dnd = (options: DNDOptions = {}): Extension => {
```

Comment:
seems like this is specifically for file dropping?

Reply:
ok. renamed.